### PR TITLE
fix: always return "completed" course run status if learner has a passing certificate

### PR DESCRIPTION
**Pull request by adamstankiewicz**
This PR modifies the `get_course_run_status` util function such that if a learner has a passing certificate, it will always return the course as "completed". This results in the following cases:

1. Enrollment saved for later --> `CourseRunProgressStatuses.SAVED_FOR_LATER`
1. Course has started but not ended --> `CourseRunProgressStatuses.IN_PROGRESS`
1. Course has ended --> `CourseRunProgressStatuses.COMPLETED`
1. Learner has passing certificate --> `CourseRunProgressStatuses.COMPLETED`
1. Course has not started --> `CourseRunProgressStatuses.UPCOMING`

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_learner_portal/utils.py [hunk 1]
```diff
@@ -21,45 +17,32 @@ class CourseRunProgressStatuses:
 
 def get_course_run_status(course_overview, certificate_info, enterprise_enrollment):
     """
-    Get the progress status of a course run, given the state of a user's certificate in the course.
+    Get the status of a course run, given the state of a user's certificate in the course.
 
-    In the case of self-paced course runs, the run is considered completed when either the course run has ended
-    OR the user has earned a passing certificate 30 days ago or longer.
```
Comment (iloveagent57) [inline enterprise_learner_portal/utils.py:27]: What was the original motivation for this 30-day wait period?

File: enterprise_learner_portal/utils.py [hunk 2]
```diff
@@ -21,45 +17,32 @@ class CourseRunProgressStatuses:
...
-            ``created``: the date the certificate was created
 
     Returns:
         status: one of (
+            CourseRunProgressStatuses.SAVED_FOR_LATER,
             CourseRunProgressStatuses.COMPLETE,
             CourseRunProgressStatuses.IN_PROGRESS,
             CourseRunProgressStatuses.UPCOMING,
         )
         None if pacing type is not matched
```
Comment (iloveagent57) [inline enterprise_learner_portal/utils.py:37]: Nit - this will no longer return null.

**Review by iloveagent57 — APPROVED** [review]
LGTM, one nit to update the docstring.

File: enterprise_learner_portal/utils.py (hunk 1 above)
Comment (adamstankiewicz) [reply enterprise_learner_portal/utils.py:27]: Good question. We tried to replicate some logic around course run status from edx-platform 17 months ago, which had that 30-day wait period built in.

This was for the `program_enrollments` endpoint. I checked with Joe about the 30-day wait period to see if there was a valid reason for it; he reached out to Marco who wasn't even aware of it haha.

So, no need for it here; it's not serving any purpose really other than causing UX issues :)

**Pull request merged**